### PR TITLE
[7.0] EZP-28262: Moved DBMS-specific SQL data file to DBMS subdirectory

### PR DESCRIPTION
**Pull request by alongosz**
| Q | A |
|---|---|
| **Issue** | [EZP-28262](https://jira.ez.no/browse/EZP-28262) |
| **Related to** | [EZP-24842](https://jira.ez.no/browse/EZP-24842) |
| **BC break** | possibly |
| **Target branch** | `7.0` |

This PR is one of the steps towards [EZP-24842](https://jira.ez.no/browse/EZP-24842).
It moves `cleandata.sql` to DBMS-specific subdirectory of `/data` and at the same time adds checking if DBMS-specific SQL files exist. 
SQL data directory can be easily changed by other installer implementations.

The Installer modified here will become of course obsolete once [EZP-24842](https://jira.ez.no/browse/EZP-24842) gets completed, however moving cleandata is a small step needed for this parent issue.

**BC**:
Method performing check is not used directly by `DbBasedInstaller` so it is not a BC break.
However I'm concerned about moving `cleandata.sql` itself - this could be seen as BC, so for now I'm targeting this to **7.0**. WDYT?

**TODO**:
- [x] **Rebase after #2157 gets merged and rebased into `7.0`**
- [x] Move cleandata to `/data/mysql` subdirectory.
- [x] Improve Installer to simplify checking existence and getting SQL data files path.

### CODE REVIEW

File: eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php [hunk 1]
```diff
@@ -67,4 +73,34 @@ protected function runQueriesFromFile($file)
...
+    /**
+     * Get DBMS-specific SQL data file path.
+     *
+     * @param string $relativeFilePath SQL file path relative to {baseDir}/{dbms} directory
+     *
+     * @return string absolute existing file path
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     */
+    protected function getDatabasePlatformSpecificSQLFile($relativeFilePath)
```
Comment (andrerom) [inline eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: maybe even set as final

Comment (andrerom) [reply eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: And maybe we should make this method available on 6.x as well, so installers can work across both versions if package maintainer wants.

Comment (bdunogier) [reply eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: That would be a good idea, yes.

**Comment by alongosz** [comment]
@andrerom: ok, this can be useful for our demo bundles. But backport it to 6.7 as well or master only?

**Comment by andrerom** [comment]
@alongosz master only

**Comment by alongosz** [comment]
@andrerom done, see #2157

**Comment by alongosz** [comment]
Just as a reminder: this was one of the steps before replacing Installer.

Rebased, [Installer works as expected](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/316875293#L940)
review ping @andrerom @Nattfarinn

**Comment by andrerom** [comment]
@Nattfarinn can we assume you are ok with this now given your comments on the other PR was fixed?

**Comment by Nattfarinn** [comment]
@andrerom Approved. :)

**Pull request merged**